parser/rootfs_parser.go

```diff
@@ -30,26 +30,25 @@ import (
 	"github.com/pkg/errors"
 )
 
+// User provided update data stream handler. Parameter `r` is a decompressed
+// data stream, `dt` holds current device type, `uf` contains basic information
+// about update. The handler shall return nil if no errors occur.
+type DataHandlerFunc func(r io.Reader, dt string, uf UpdateFile) error
+
 type RootfsParser struct {
-	dataWriter io.Writer
-	scriptDir  string
+	W         io.Writer
```

Comment:
Trying to follow the naming pattern established by stdlib, for instance: https://godoc.org/io#LimitedReader

I'll add a comment that W is where data is written to.